instance/models/instance.py

```diff
@@ -0,0 +1,436 @@
+    """
+    Instances linked to a VCS, such as git
+    """
+
+    class Meta:
+        abstract = True
+
+    branch_name = models.CharField(max_length=50, default='master')
+    ref_type = models.CharField(max_length=50, default='heads')
+    commit_id = models.CharField(max_length=40, validators=[sha1_validator])
+
+    @property
+    def commit_short_id(self):
+        """
+        Short `commit_id`, limited to 7 characters like on GitHub
+        """
+        if not self.commit_id:
+            return None
+        return self.commit_id[:7]
+
+
+class GitHubInstanceMixin(VersionControlInstanceMixin):
+    """
+    Instance linked to a GitHub repository
+    """
+    github_organization_name = models.CharField(max_length=50, db_index=True)
+    github_repository_name = models.CharField(max_length=50, db_index=True)
+    github_admin_organization_name = models.CharField(max_length=50, blank=True,
+                                                      default=settings.DEFAULT_ADMIN_ORGANIZATION)
+
+    class Meta:
+        abstract = True
+
+    @property
+    def fork_name(self):
+        """
+        Fork name (eg. 'open-craft/edx-platform')
+        """
+        return '{0.github_organization_name}/{0.github_repository_name}'.format(self)
+
+    @property
+    def github_base_url(self):
+        """
+        Base GitHub URL of the fork (eg. 'https://github.com/open-craft/edx-platform')
+        """
+        return 'https://github.com/{0.fork_name}'.format(self)
+
+    @property
+    def github_branch_url(self):
+        """
+        GitHub URL of the branch tree
+        """
+        return '{0.github_base_url}/tree/{0.branch_name}'.format(self)
+
+    @property
+    def repository_url(self):
+        """
+        URL of the git repository (eg. 'https://github.com/open-craft/edx-platform.git')
+        """
+        return '{0.github_base_url}.git'.format(self)
+
+    @property
+    def updates_feed(self):
+        """
+        RSS/Atom feed of commits made on the repository/branch
+        """
+        return '{0.github_base_url}/commits/{0.branch_name}.atom'.format(self)
+
+    @property
+    def github_admin_username_list(self):
+        """
+        Returns the github usernames of this instance admins
+
+        Admins are the members of the default team of the `github_admin_organization_name` org
+        """
+        if self.github_admin_organization_name:
+            return get_username_list_from_team(self.github_admin_organization_name)
+        else:
+            return []
+
+    def set_to_branch_tip(self, branch_name=None, ref_type=None, commit=True):
+        """
+        Set the `commit_id` to the current tip of the branch
+        """
+        if branch_name is not None:
+            self.branch_name = branch_name
+        if ref_type is not None:
+            self.ref_type = ref_type
+        self.log('info', 'Setting instance {} to tip of branch {}'.format(self, self.branch_name))
```

Comment:
@e-kolpakov I'd rather not use `logging.INFO` for this - the value is stored in the DB, and besides the fact that it would expose a logging interna, that's an integer and would be stored as such in the DB (ie, slimmer, but less explicit). Switching to custom constants works for me though, good point.